libraries/AP_Motors/AP_MotorsTailsitter.cpp

```diff
@@ -85,28 +84,29 @@ void AP_MotorsTailsitter::output_to_motors()
             break;
         }
     }
-    // outputs are setup here, and written to the HAL by the plane servos loop
-    SRV_Channels::set_output_scaled(SRV_Channel::k_aileron,  _aileron*SERVO_OUTPUT_RANGE);
-    SRV_Channels::set_output_scaled(SRV_Channel::k_elevator, _elevator*SERVO_OUTPUT_RANGE);
-    SRV_Channels::set_output_scaled(SRV_Channel::k_rudder,   _rudder*SERVO_OUTPUT_RANGE);
-    SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, throttle*THROTTLE_RANGE);
-
-    // also support differential roll with twin motors
+    // roll with thrust differential from twin motors
     SRV_Channels::set_output_scaled(SRV_Channel::k_throttleLeft,  throttle_left*THROTTLE_RANGE);
     SRV_Channels::set_output_scaled(SRV_Channel::k_throttleRight, throttle_right*THROTTLE_RANGE);
 
+    // thrust vectoring
+    SRV_Channels::set_output_scaled(SRV_Channel::k_tiltMotorLeft, _tilt_left*SERVO_OUTPUT_RANGE);
+    SRV_Channels::set_output_scaled(SRV_Channel::k_tiltMotorRight, _tilt_right*SERVO_OUTPUT_RANGE);
+
+    // plane outputs for Qmodes are setup here, and written to the HAL by the plane servos loop
+    SRV_Channels::set_output_scaled(SRV_Channel::k_aileron,  -_yaw_in*SERVO_OUTPUT_RANGE);
```

Comment:
It's existing code so it's OK for this PR but it's a bit weird that the motors library is setting the output ranges for k_aileron, k_elevator, k_rudder and k_throttle.  I would much prefer if one part of the code was responsible for the setup and output of each servo/esc output rather than having this responsibility done in two very separate parts of the code.

Reply:
Yeah, i think this is just a peculiarity of how the tail-sitter code works as the same output functions are used by both the copter and plane code. Depending on if its in a forward flight mode these outputs are overwritten.